app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt

```diff
@@ -23,10 +33,70 @@ class AmortisationsrechnerFragment : Fragment() {
         amortisationsrechnerViewModel =
             ViewModelProviders.of(this).get(com.example.stromtracker.ui.amortisationsrechner.AmortisationsrechnerViewModel::class.java)
         val root = inflater.inflate(R.layout.fragment_amortisationsrechner, container, false)
-        val textView: TextView = root.findViewById(R.id.text_amortisationsrechner)
-        amortisationsrechnerViewModel.text.observe(viewLifecycleOwner, Observer {
-            textView.text = it
-        })
+
+        editVerbrauchAkt = root.findViewById(R.id.amort_edit_verbrauch_akt)
+        editVerbrauchNeu = root.findViewById(R.id.amort_edit_verbrauch_neu)
+        editAnschaffung = root.findViewById(R.id.amort_edit_anschaffung)
+        editStrompreis = root.findViewById(R.id.amort_edit_strompreis)
+
+
+        outKostenAkt = root.findViewById(R.id.amort_text_kosten_akt_num)
+        outKostenNeu = root.findViewById(R.id.amort_text_kosten_neu_num)
+        outAmortdauer = root.findViewById(R.id.amort_text_amortdauer_num)
+
+        addCustomTextChangedListener(editVerbrauchAkt)
+        addCustomTextChangedListener(editVerbrauchNeu)
+        addCustomTextChangedListener(editAnschaffung)
+        addCustomTextChangedListener(editStrompreis)
+
         return root
     }
+
+    fun addCustomTextChangedListener(edit:EditText) : EditText {
+        edit.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable) {
+                val verbrAkt : Double? = editVerbrauchAkt.text.toString().toDoubleOrNull()
```

Comment:
This method is quite long. Could be separated into smaller methods. If this is commonly used from different UI elements it makes definitely sense to separate it from the UI.

Reply:
This has been fixed in Amortisationsrechner Pull Request. Merged it in here, so it's fixed.